tracer/transport_test.go

```diff
@@ -24,30 +24,38 @@ func getTestSpan() *Span {
 	}
 }
 
-// getTestTrace returns a []Span that is composed by ``size`` number of spans
-func getTestTrace(size int) []*Span {
-	spans := []*Span{}
-
-	for i := 0; i < size; i++ {
-		spans = append(spans, getTestSpan())
+// getTestTrace returns a list of traces that is composed by ``traceN`` number
+// of traces, each one composed by ``size`` number of spans.
+func getTestTrace(traceN, size int) [][]*Span {
+	var traces [][]*Span
+
+	for i := 0; i < traceN; i++ {
+		trace := []*Span{}
+		for j := 0; j < size; j++ {
+			trace = append(trace, getTestSpan())
+		}
+		traces = append(traces, trace)
 	}
-	return spans
+	return traces
 }
 
 func TestTracesAgentIntegration(t *testing.T) {
 	assert := assert.New(t)
 
 	testCases := []struct {
-		payload []*Span
+		payload [][]*Span
 	}{
-		{getTestTrace(1)},
-		{getTestTrace(10)},
+		{getTestTrace(1, 1)},
```

Comment:
Tests all combination of payload (single-trace/single-span, multiple-trace/single-span, etc...). Useful later with ``msgpack``